cgp/ea/mu_plus_lambda.py

```diff
@@ -142,6 +142,8 @@ def step(
         # concatenating the parent population to the offspring
         # population instead of the other way around
         combined = offsprings + pop.parents
+        assert all(ind.idx is not None for ind in combined)
+        assert len(set(ind.idx for ind in combined)) == len(combined)
```

Comment:
Maybe below could be easier to understand? (No preference though, both ways should be fine)
```python
assert len(set(ind.idx for ind in combined)) == len([int.idx for ind in combined])
```